graylog2-web-interface/src/routing/context/ScratchpadProvider.jsx

```diff
@@ -0,0 +1,27 @@
+
+import React, { createContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const ScratchpadContext = createContext();
+
+export const ScratchpadProvider = ({ children }) => {
```

Comment:
I'm wondering if there's a better place to put these providers. I understand we use the `ScratchpadProvider` in routing and it probably makes sense having it here, but `ScratchpadContext` gets used in `Scratchpad`. I don't have a better suggestion for this at the moment, but still wanted to mention in case you think it could make sense having providers (in general) somewhere else.